python/cudf/cudf/core/column/string.py

```diff
@@ -4706,7 +4706,7 @@ def normalize_characters(self, do_lower: bool = True) -> SeriesOrIndex:
         dtype: object
         """
         return self._return_or_inplace(
-            self._column.normalize_characters(do_lower)
+            self._column.characters_normalize(do_lower)
```

Comment:
So `characters_normalize` is the old API which will be deprecated in a follow-up PR?

Reply:
Yes. Is there an official way to communicate that in the comments perhaps?

Reply:
I think maybe want to to output a warning using [warnings](https://docs.python.org/3/library/warnings.html). And then we might have to make the tests which use this API expect a warning because we treat warnings as errors. If this is the right approach, I can also handle this in a follow-up PR.

cc @vyasr Do we want to output a warning when deprecating this API?

Reply:
Added deprecation warning.

Reply:
I don't think this API is used anywhere else in RAPIDS, so adding a warning here shouldn't break CI.